packages/config/scripts/diff/diff.ts

```diff
@@ -0,0 +1,159 @@
+import { type Change, diffLines } from 'diff'
+import type { Project } from './types'
+
+export interface ProjectDiff {
+  id: string
+  type: 'added' | 'removed' | 'modified'
+  fields: FieldDiff[]
+  added: number
+  removed: number
+}
+
+export interface FieldDiff {
+  field: string
+  diff: LineChange[]
+  added: number
+  removed: number
+}
+
+export interface LineChange {
+  type: 'full' | 'ctx' | 'added' | 'removed'
+  value: string
+}
+
+export function diffAll(
+  projectsBefore: Project[],
+  projectsAfter: Project[],
+): ProjectDiff[] {
+  const result: ProjectDiff[] = []
+
+  let beforeIndex = 0
+  let afterIndex = 0
+  while (
+    beforeIndex < projectsBefore.length ||
+    afterIndex < projectsAfter.length
+  ) {
+    const before = projectsBefore[beforeIndex]
+    const after = projectsAfter[afterIndex]
+
+    if (!before) {
```

Comment:
What will happen if in the following scenario

```
| before projects | after projects |
|-----------------|----------------|
|        arbitrum |       arbitrum |
|        starknet |                |
|            zora |           zora |
```

We want to see `starknet` as removed but since we iterate over indexes we're going to show a diff for starknet between it and zora plus later we're going to say that zora was removed. Or am i mistaken?